dissect/target/plugins/apps/webservers/caddy.py

```diff
@@ -0,0 +1,128 @@
+        if default_log_file.exists():
+            log_paths.append(default_log_file)
+
+        # Check for custom paths in Caddy config
+        if (config_file := self.target.fs.path("/etc/caddy/Caddyfile")).exists():
+            found_roots = []
+            for line in config_file.open("rt"):
+                line = line.strip()
+
+                if not line or line.startswith("#"):
+                    continue
+
+                if "root " in line:
+                    found_roots.append(line.split("root ")[1].strip())
+
+                if "log " in line or "output file " in line:
+                    if line.strip() == "log {":
+                        continue
+
+                    match = LOG_FILE_REGEX.match(line)
+                    if not match:
+                        self.target.log.warning("Could not determine log path in line %s", line)
+                        continue
+
+                    match = match.groupdict()
+                    p = match["log_file"].replace(" {", "").split(" ")[-1]
+
+                    if not p.startswith("/"):
+                        print(f"{p} is a relative path")
+                        for root in found_roots:
+                            root_parent = self.target.fs.path(f"{root}").parent
+                            abs_log_path = self.target.fs.path(f"{str(root_parent)}/{p}").parent
+                            print(abs_log_path)
+                            if abs_log_path.exists() and abs_log_path not in log_paths:
+                                log_paths.append(abs_log_path)
+                    else:
+                        print(f"{p} is an absolute path")
+                        if (path := self.target.fs.path(p).parent).exists():
+                            if path not in log_paths:
+                                log_paths.append(path)
+
+        return log_paths
+
+    def check_compatible(self) -> bool:
+        return len(self.log_paths) > 0
+
+    @plugin.export(record=WebserverRecord)
+    def access(self) -> Iterator[WebserverRecord]:
+        """Parses Caddy V1 CRF and Caddy V2 JSON logs.
+
+        Resources:
+            - https://caddyserver.com/docs/caddyfile/directives/log#format-modules
+        """
+        tzinfo = self.target.datetime.tzinfo
+
+        for path in self.log_paths:
+            for line in open_decompress(path, "rt"):
+                line = line.strip()
+                if not line:
+                    continue
+
+                # Parse a JSON log line
+                if line.startswith('{"'):
+                    try:
+                        log = json.loads(line)
+                    except json.decoder.JSONDecodeError:
+                        self.target.log.warning("Could not decode Caddy JSON log line in file %s", path)
+                        continue
+
+                    yield WebserverRecord(
+                        ts=from_unix(log["ts"]).replace(tzinfo=tzinfo),
```

Comment:
```suggestion
                        ts=from_unix(log["ts"]),
```
I assume `Caddy` logs true Unix (i.e. UTC) timestamps in this field? So no need to transform to another timezone.

Reply:
Unfortunately this does not seem to be the case. The epoch timestamp seems to be relative to the system's time.
```
FAILED tests/test_plugins_apps_webservers_caddy.py::test_plugins_apps_webservers_caddy_json - AssertionError: assert 2023-02-06 15:05:49.064393+00:00 == 2023-02-06 15:05:49.064393+01:00
```

Reply:
Re-added in [4e5da15](https://github.com/fox-it/dissect.target/pull/127/commits/4e5da1558e3db1b9c9c76ac1188422b3442c4c71).